src/number-input/NumberInputDecrementButton.ts

```diff
@@ -0,0 +1,67 @@
+import { useCallback } from "react";
+import { createComponent, createHook } from "reakit-system";
+import { ariaAttr, callAllHandlers } from "@chakra-ui/utils";
+import { ButtonHTMLProps, ButtonOptions, useButton } from "reakit/Button";
+
+import { NumberInputStateReturn } from "./NumberInputState";
+import { NUMBERINPUT_DECREMENTBUTTON_KEYS } from "./__keys";
+
+export type NumberInputDecrementButtonOptions = ButtonOptions &
+  Pick<Partial<NumberInputStateReturn>, "keepWithinRange"> &
+  Pick<
+    NumberInputStateReturn,
+    "focusInput" | "increment" | "decrement" | "isAtMin" | "spinner"
+  >;
+
+export type NumberInputDecrementButtonHTMLProps = ButtonHTMLProps;
+
+export type NumberInputDecrementButtonProps = NumberInputDecrementButtonOptions &
+  NumberInputDecrementButtonHTMLProps;
```

Comment:
Unused Type

Reply:
I am doing it with other fixes !!
Expect a commit soon.

Reply:
Hey, just realised. This is not an unused Type.

This is used when we use this component as hooks, You can visit any of the reakit components and will see this props exported.

Useful to type the hook's props for the users who use it IMO.
Not really used within the components.

Reply:
Ohhhh i see, okay okay got it!